src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -0,0 +1,575 @@
+		if player == Players.LocalPlayer then
+			-- Before player leaves the game,
+			-- log the number of times the shop was opened
+			AnalyticsService.LogEvent:Fire({ -- Use AnalyticsService.LogEvent:Fire() to log a client-side event
+				event: "UIEvent:OpenedShop",
+				value: timesOpenedShop -- Number of times the player opened their shop
+			})
+		end
+	end)
+	```
+]=]
+local AnalyticsService = Knit.CreateService({
+	Name = "AnalyticsService";
+	Client = {
+		LogEvent = Knit.CreateSignal();
+	};
+})
+
+--[=[
+	@interface PlayerEvent
+	.userId number
+	.event string
+	.value number?
+	@within AnalyticsService
+]=]
+type PlayerEvent = {
+	userId: number,
+	event: string,
+	value: number?
+}
+
+--[=[
+	@interface MarketplacePurchaseEvent
+	.userId number
+	.itemType string
+	.id string
+	.amount number?
+	.cartType string
+	@within AnalyticsService
+]=]
+type MarketplacePurchaseEvent = {
+	userId: number,
+	itemType: string,
+	id: string,
+	amount: number?,
+	cartType: string
+}
+
+--[=[
+	@interface PurchaseEvent
+	.userId number
+	.eventType string
+	.itemId string
+	.currency string
+	.flowType string? -- Allowed flow types: "Sink", "Source" (defaults to "Sink")
+	.amount number?
+	@within AnalyticsService
+	
+	- Currency is the in-game currency type used, it must be defined in the CURRENCIES table
+]=]
+type PurchaseEvent = {
+	userId: number,
+	eventType: string, -- 1 by default
+	itemId: string,
+	currency: string, -- In-game currency type used
+	flowType: string?, -- Sink or Source (Sink by default)
+	amount: number?
+}
+
+--[=[
+	@interface ResourceEvent
+	.userId number
+	.eventType string
+	.itemId string
+	.currency string
+	.flowType string
+	.amount number
+	@within AnalyticsService
+	
+	- Currency is the in-game currency type used, it must be defined in the CURRENCIES table
+]=]
+type ResourceEvent = {
+	userId: number,
+	eventType: string,
+	itemId: string, -- unique id of item (example: "Coins", "100 Coins",
+	-- "Coin Pack", "Weapon Skin", "Red Loot Box", "Extra Life")
+	currency: string,
+	flowType: string,
+	amount: number
+}
+
+--[=[
+	@interface ErrorEvent
+	.message string
+	.severity string? -- Allowed severities: "Debug", "Info", "Warning", "Error", "Critical" (defaults to "Error")
+	.userId number
+	@within AnalyticsService
+]=]
+type ErrorEvent = {
+	message: string,
+	severity: string?,
+	userId: number
+}
+
+--[=[
+	@interface ProgressionEvent
+	.userId number
+	.status string -- Allowed statuses: "Start", "Fail", "Complete"
+	.progression01 string
+	.progression02 string?
+	.progression03 string?
+	.score number?
+	@within AnalyticsService
+]=]
+type ProgressionEvent = {
+	userId: number,
+	status: string, -- Start, Fail, Complete
+	progression01: string, -- Mission, Level, etc.
+	progression02: string?, -- Location, etc.
+	progression03: string?, -- Level, etc. (if used then progression02 is required)
+	score: number?
+}
+
+function AnalyticsService:KnitInit()
+	GameAnalytics:configureBuild(BUILD)
```

Comment:
Consider moving the configuration-related functions to the KnitStart function of this service - that way other scripts can set variables under this service during the init phase.